packages/aws-rfdk/lib/core/lib/mountable-efs.ts

```diff
@@ -157,11 +169,26 @@ export class MountableEfs implements IMountableLinuxFilesystem {
     }
     const mountOptionsStr: string = mountOptions.join(',');
 
+    const resolveMountTargetDnsWithApi = this.props.resolveMountTargetDnsWithApi ?? false;
+    if (resolveMountTargetDnsWithApi) {
+      const describeMountTargetResources = [
+        (this.props.filesystem.node.defaultChild as efs.CfnFileSystem).attrArn,
+      ];
+      if (this.props.accessPoint) {
+        describeMountTargetResources.push(this.props.accessPoint.accessPointArn);
+      }
+
+      target.grantPrincipal.addToPrincipalPolicy(new PolicyStatement({
+        resources: describeMountTargetResources,
+        actions: ['elasticfilesystem:DescribeMountTargets'],
+      }));
+    }
+
     target.userData.addCommands(
       'TMPDIR=$(mktemp -d)',
       'pushd "$TMPDIR"',
       `unzip ${mountScript}`,
-      `bash ./mountEfs.sh ${this.props.filesystem.fileSystemId} ${mountDir} ${mountOptionsStr}`,
+      `bash ./mountEfs.sh ${this.props.filesystem.fileSystemId} ${mountDir} ${mountOptionsStr} ${resolveMountTargetDnsWithApi}`,
```

Comment:
I suggest reversing the ordering of mount options & the new arg. Reason being that mount options could be empty, but the new arg will never be.

Reply:
done